src/routes/safe/components/Apps/components/SignMessageModal/index.tsx

```diff
@@ -39,19 +42,45 @@ const convertToHumanReadableMessage = (message: string): string => {
   return humanReadableMessage
 }
 
-export const SignMessageModal = ({ message, isOpen, ...rest }: SignMessageModalProps): ReactElement => {
+export const SignMessageModal = ({ message, isOpen, method, ...rest }: SignMessageModalProps): ReactElement => {
   const web3 = getWeb3ReadOnly()
   const networkId = useSelector(currentChainId)
   const txRecipient = getSignMessageLibAddress(networkId) || ZERO_ADDRESS
-  const txData = getSignMessageLibContractInstance(web3, networkId)
-    .methods.signMessage(web3.eth.accounts.hashMessage(message))
-    .encodeABI()
-
-  const readableData = convertToHumanReadableMessage(message)
+  let txData, readableData
+  if (method == Methods.signMessage) {
+    txData = getSignMessageLibContractInstance(web3, networkId)
+      .methods.signMessage(web3.eth.accounts.hashMessage(message))
+      .encodeABI()
+    readableData = convertToHumanReadableMessage(message)
+  } else if (method == Methods.signTypedMessage) {
+    // Here we firstly convert the typed message to safe typed message which contains the safe address and the current chainId,
+    // then we hash it using ethers.utils._TypedDataEncoder.
+    txData = getSignMessageLibContractInstance(web3, networkId)
+      .methods.signMessage(
+        _TypedDataEncoder.hash(
+          { verifyingContract: rest.safeAddress, chainId: parseInt(networkId) },
```

Comment:
What's the type of networkId? I believe `ethers.js` can accept both string and number

Reply:
Yes, `ethers.js` can. My intention is to align this line of code with the encoding in `safe-apps-sdk`, where the `chainId` is a `number`:

https://github.com/safe-global/safe-apps-sdk/pull/356/files#diff-cd9509515879465c31ec3312568d5248bb8606bf20a7e9e6be3ee8d373ed7185R125

Reply:
Understood, thanks. The resulting hash should be the same no matter what, so there's no real need to do it, but also not a big deal